### PR TITLE
Fixing content serializer bugs

### LINKED ISSUE #575: Invalid hyperlink when creating ReleaseComponent
Hoping someone can confirm whether this is a bug or maybe I am doing something wrong?

**Version**
pulpcore 3.19.0
pulp_deb 2.18.0

**Describe the bug**
I am trying to create a ReleaseComponent but getting `Invalid hyperlink - Incorrect URL match.` for the release field.

**To Reproduce**
```
RELEASE_HREF=$(http :8080/pulp/api/v3/content/deb/releases/ codename=mycodename$RANDOM suite=noidea$RANDOM distribution=mydist$RANDOM | jq -r ".pulp_href")
echo $RELEASE_HREF
http :8080/pulp/api/v3/content/deb/release_components/ component=mycomp release=$RELEASE_HREF
```

Response:

```
{
    "release": [
        "Invalid hyperlink - Incorrect URL match."
    ]
}
```

**Expected behavior**
The component is created.

### PR DESCRIPTION
**Pull request by daviddavis**
fixes #575

### CODE REVIEW

**Comment by daviddavis** [comment]
Test script:

```
URL=":8080"

RELEASE_HREF=$(http ${URL}/pulp/api/v3/content/deb/releases/ codename=mycodename$RANDOM suite=noidea$RANDOM distribution=mydist$RANDOM | jq -r ".pulp_href")
echo $RELEASE_HREF

COMP_HREF=$(http ${URL}/pulp/api/v3/content/deb/release_components/ component=mycomp release=$RELEASE_HREF | jq -r ".pulp_href")
echo $COMP_HREF

wget https://fixtures.pulpproject.org/debian/pool/asgard/f/frigg/frigg_1.0_ppc64.deb
TASK_HREF=$(http --form $URL/pulp/api/v3/content/deb/packages/ file@frigg_1.0_ppc64.deb | jq -r ".task")
PACKAGE_HREF=$(http ${URL}${TASK_HREF} | jq -r ".created_resources[0]")
echo $PACKAGE_HREF

http ${URL}/pulp/api/v3/content/deb/package_release_components/ package=$PACKAGE_HREF release_component=$COMP_HREF

http ${URL}/pulp/api/v3/content/deb/release_architectures/ architecture=$RANDOM release=$RELEASE_HREF
```

File: pulp_deb/app/serializers/content_serializers.py [hunk 1]
```diff
@@ -617,7 +617,7 @@ class ReleaseArchitectureSerializer(NoArtifactContentSerializer):
         help_text="Release this architecture is contained in.",
         many=False,
         queryset=Release.objects.all(),
-        view_name="deb-release-detail",
+        view_name="content-deb/releases-detail",
```
Comment (quba42) [inline pulp_deb/app/serializers/content_serializers.py:620]: Can you explain to me what the significance of the view name changes is? I have never really touched those before and just assumed they are fine the way they are and used for god knows what...

File: CHANGES/575.bugfix [hunk 2]
```diff
@@ -0,0 +1 @@
+Fixed bugs when creating content such as releases, comps, etc.
```
Comment (quba42) [inline CHANGES/575.bugfix:1]:
```suggestion
Fixed several serializer bugs preventing the manual creation of structure content of type ``ReleaseArchitecture``, ``ReleaseComponent``, and ``PackageReleaseComponent``.
```

**Review by quba42 — CHANGES_REQUESTED** [review]
I have a suggestion for the change log entry. I will now try to test the intended workflow locally. Assuming that goes well, this all LGTM. Thanks for looking into this!

**Comment by quba42** [comment]
I was able to verify the fix works. By choosing meaningful architecture, component, and distribution fields, I was even able to create a valid structured publication + distribution!

File: pulp_deb/app/serializers/content_serializers.py (hunk 1 above)
Comment (daviddavis) [reply pulp_deb/app/serializers/content_serializers.py:620]: I'm not an expert here (I think maybe @mdellweg might know more) but here's what I know: This is a `DetailRelatedField` field [which extends DRF's `HyperlinkedRelatedField`](https://github.com/pulp/pulpcore/blob/c2c4e28477d2b9ee6a038680adafe76e0b53a5f8/pulpcore/app/serializers/base.py#L332) (more info on HyperlinkedRelatedField and `view_name` [in the DRF docs here](https://www.django-rest-framework.org/api-guide/relations/#hyperlinkedrelatedfield)). I think `view_name` needs to match the name of the route (which I found with `pulpcore-manager show_urls`). I believe DRF uses this to match the href and extract the info it needs from the href.

Comment (mdellweg) [reply pulp_deb/app/serializers/content_serializers.py:620]: The change looks good to me. At the time i wrote this code, i think i was guessing what the view would be named. Also there might have been an architectural change in pulpcore back in the days that reshuffled all the detail view names.

**Pull request merged**